WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsViewModel.swift

```diff
@@ -338,13 +338,33 @@ private extension PaymentMethodsViewModel {
     ///
     @MainActor
     func markOrderAsPaid() async throws {
-        try await withCheckedThrowingContinuation { continuation in
-            stores.dispatch(OrderAction.updateOrderStatus(siteID: siteID, orderID: orderID, status: .completed) { error in
-                guard let error else {
-                    return continuation.resume(returning: ())
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            guard let order = ordersResultController.fetchedObjects.first else {
+                DDLogError("⛔️ Order not found, can't mark order as paid.")
+                continuation.resume(throwing: PaymentMethodsError.orderNotFound)
+                return
+            }
+
+            let cashOnDeliveryID = PaymentGateway.Constants.cashOnDeliveryGatewayID
```

Comment:
I have my concerns hardcoding these title and payment gateway here. After all, this method is just marking the order as paid (complete); it doesn't have to be necessarily cash on delivery. I would change the function name to make it more explicit (e.g. `markOrderCompleteWithCashOnDelivery`), or pass the method id and title as parameters :)

Reply:
Good point. This method is only used for Cash Payments (Order -> Collect Payment -> Cash) flow, and is only called by a larger `markOrderAsPaidByCash` method. So I agree the naming is too broad and misleading. 👍

Reply:
Something like `completeCashPayment` or `completeCashOnDeliveryPayment` could reflect the functionality in a most appropriate way.